### DISCUSSION TITLE
iOS cert_path: StringIO and File do not work

### DISCUSSION THREAD

**Discussion by chagar** [excid3/noticed#332] [General]
* PR #183 purported to add support for StringIO and File types in iOS `cert_path`, but it does not work.
* Error message:
    ```
    ArgumentError: Could not find APN cert at '[removed]/config/certs/ios/apns.p8'
    ```
* This is the default path for the cert file, from `Noticed::DeliveryMethods::Ios#cert_path`. It reaches the default path because this method checks for a `String` or a `Symbol`, but not a `StringIO` or `File` object. Only the validation was changed in the PR.
* Adding `StringIO` next to `String` in the case statement makes it work.
* Likewise, `File.open` does not work either, but adding `File` to `#cert_path` makes it work.
* Simple fix:
    ```ruby
      def cert_path    
        option = options[:cert_path]    
        case option    
        when String, StringIO, File    
          option    
        when Symbol    
          notification.send(option)    
        else    
          Rails.root.join("config/certs/ios/apns.p8")    
        end    
      end
    ```
* However, since cert_path is merely being passed to apnotic, which accepts any object that responds to `:read`, it may make more sense to fit the noticed code to apnotic. See:
    * https://github.com/ostinelli/apnotic/blob/master/lib/apnotic/connection.rb#L30
    * https://github.com/ostinelli/apnotic/blob/master/lib/apnotic/connection.rb#L130

**Comment by excid3** [comment]
PRs welcome. 👍

**Comment by chagar** [comment]
@excid3 Is anything else needed to get this merged? (PR #206) Should I keep updating it?
